### PR TITLE
IDVA3-1307 create service to retrieve company individual pscs

### PR DESCRIPTION
**Pull request by AlisonGriffiths**

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: test/services/companyPscService.unit.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    });
+
+    it("getCompanyPscList should return 200 OK HttpStatus response", async () => {
+        const mockResponse: ApiResponse<CompanyPersonsWithSignificantControlResource> = {
+            httpStatusCode: HttpStatusCode.Ok,
+            resource: VALID_COMPANY_PSC_LIST
+        };
+        mockGetCompanyPsc.mockResolvedValueOnce(mockResponse as ApiResponse<CompanyPersonsWithSignificantControlResource>);
+        const request = {} as Request;
+        // request.query = { companyNumber: COMPANY_NUMBER };
```
Comment (JoelClemence) [inline test/services/companyPscService.unit.ts:36]: Is this needed?

File: test/services/companyPscService.unit.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    });
+    it("getCompanyPscList should throw an error if HttpStatus is not 200 OK", async () => {
+        const mockResponse: ApiResponse<CompanyPersonsWithSignificantControlResource> = {
+            httpStatusCode: HttpStatusCode.ServiceUnavailable
+        };
+        mockGetCompanyPsc.mockResolvedValueOnce(mockResponse as ApiResponse<CompanyPersonsWithSignificantControlResource>);
+        const request = {} as Request;
+
+        try {
+            const response = await getCompanyPscList(request, COMPANY_NUMBER);
```
Comment (JoelClemence) [inline test/services/companyPscService.unit.ts:56]: Fail if no error thrown?

File: test/services/companyPscService.unit.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        }
+    });
+    it("getCompanyPscList should throw an error if no resource is returned", async () => {
+        const mockResponse: ApiResponse<CompanyPersonsWithSignificantControlResource> = {
+            httpStatusCode: HttpStatusCode.Ok
+        };
+        mockGetCompanyPsc.mockResolvedValueOnce(mockResponse as ApiResponse<CompanyPersonsWithSignificantControlResource>);
+        const request = {} as Request;
+        try {
+            await getCompanyPscList(request, COMPANY_NUMBER);
```
Comment (JoelClemence) [inline test/services/companyPscService.unit.ts:68]: Again fail if no error thrown?

File: test/services/companyPscService.unit.ts (hunk 2 above)
Comment (mwejuli-ch) [reply test/services/companyPscService.unit.ts:56]: After line 56, we can add:

`throw new Error("did not get here");`

**Review by mwejuli-ch — APPROVED** [review]

File: test/services/companyPscService.unit.ts (hunk 3 above)
Comment (mwejuli-ch) [reply test/services/companyPscService.unit.ts:68]: After line 68, we can add:

`throw new Error("did not get here");`

**Review by mwejuli-ch — CHANGES_REQUESTED** [review]

File: test/services/companyPscService.unit.ts (hunk 2 above)
Comment (AlisonGriffiths) [reply test/services/companyPscService.unit.ts:56]: Thanks for the feedback, I will make those changes.

File: test/services/companyPscService.unit.ts (hunk 1 above)
Comment (AlisonGriffiths) [reply test/services/companyPscService.unit.ts:36]: I missed that one, I will remove

**Review by PhilipGlover — APPROVED** [review]
- have built and run successfully
- tests pass, coverage 93%
  - `companyPscService.ts   |   92.59 |   90.9 |   100 |    92 | 16-17`
- added `const response = await getCompanyPscList(req, "00006400");`. into `individualPscListRouter.get`
  - all PSCs returned when viewed in debug 
- added `const response = await getCompanyIndividualPscList(req, "00006400");`
  - 3 individual Pscs returned"
Screenshot 2024-05-03 at 8 08 19 AM

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by mwejuli-ch — APPROVED** [review]

File: test/services/companyPscService.unit.ts (hunk 1 above)
Comment (AlisonGriffiths) [reply test/services/companyPscService.unit.ts]: I have removed that now Joel

**Pull request merged**